core/src/main/java/io/undertow/server/HttpServerExchange.java

```diff
@@ -1397,7 +1397,7 @@ public StreamSinkChannel getResponseChannel() {
             final WrapperStreamSinkConduitFactory factory = new WrapperStreamSinkConduitFactory(wrappers, responseWrapperCount, this, sinkChannel.getConduit());
             sinkChannel.setConduit(factory.create());
         } else {
-            sinkChannel.setConduit(connection.getSinkConduit(this, sinkChannel.getConduit()));
+            sinkChannel.setConduit(new ExchangeLifecycleDetachableStreamSinkConduit(connection.getSinkConduit(this, sinkChannel.getConduit()), this));
```

Comment:
In this case there are no wrappers, so we don't necessarily need to wrap the conduit, however I opted to include the wrapper here for symmetry/safety. Please let me know if you'd prefer for this line to be reverted!